python/ray/dag/compiled_dag_node.py

```diff
@@ -1542,6 +1562,18 @@ def run(self):
         monitor.start()
         return monitor
 
+    def raise_if_too_many_inflight_requests(self):
+        num_in_flight_requests = self._execution_index - self._max_execution_index
+        if num_in_flight_requests > self._max_buffered_inputs:
```

Comment:
The comparison would be more involved:
max_num_in_flight_requests is not max_buffered_inputs, but also related to the number of channels.

See the other comment as well.

Reply:
it is cleaner with a new name `max_num_in_flight_requests`